cylc/flow/data_store_mgr.py

```diff
@@ -1770,38 +1786,54 @@ def store_node_fetcher(
                 f'{ID_DELIM}{name}{ID_DELIM}{sub_num}'
             )
             node_type = JOBS
-        if node_id in self.data[self.workflow_id][node_type]:
-            return (node_id, self.data[self.workflow_id][node_type][node_id])
-        elif node_id in self.added[node_type]:
+        if node_id in self.added[node_type]:
             return (node_id, self.added[node_type][node_id])
+        elif node_id in self.data[self.workflow_id][node_type]:
+            return (node_id, self.data[self.workflow_id][node_type][node_id])
         return (node_id, False)
 
-    def apply_deltas(self, reloaded=False):
+    def batch_deltas(self, reloaded=False):
         """Gather and apply deltas."""
         # Gather cumulative update element
-        for key, elements in self.added.items():
-            if elements:
-                if key == WORKFLOW:
-                    if elements.ListFields():
-                        self.deltas[WORKFLOW].added.CopyFrom(elements)
-                    continue
-                self.deltas[key].added.extend(elements.values())
-        for key, elements in self.updated.items():
-            if elements:
-                if key == WORKFLOW:
-                    if elements.ListFields():
-                        self.deltas[WORKFLOW].updated.CopyFrom(elements)
-                    continue
-                self.deltas[key].updated.extend(elements.values())
+        if reloaded:
+            for key, elements in self.data[self.workflow_id].items():
+                if elements:
+                    if key == WORKFLOW:
+                        if elements.ListFields():
+                            self.deltas[WORKFLOW].added.CopyFrom(elements)
+                        continue
+                    self.deltas[key].added.extend(elements.values())
+        else:
+            for key, elements in self.added.items():
+                if elements:
+                    if key == WORKFLOW:
+                        if elements.ListFields():
+                            self.deltas[WORKFLOW].added.CopyFrom(elements)
+                        continue
+                    self.deltas[key].added.extend(elements.values())
+            for key, elements in self.updated.items():
+                if elements:
+                    if key == WORKFLOW:
+                        if elements.ListFields():
+                            self.deltas[WORKFLOW].updated.CopyFrom(elements)
+                        continue
+                    self.deltas[key].updated.extend(elements.values())
```

Comment:
@dwsutherland I think these three blocks of code could be simplified. This part is always the same I think?

```py
            for key, elements in ??????:
                if elements:
                    if key == WORKFLOW:
                        if elements.ListFields():
                            self.deltas[WORKFLOW].?{updated|added}?.CopyFrom(elements)
                        continue
                    self.deltas[key].?{updated|added}?.extend(elements.values())
```

Maybe a function like `gather_delta_elements(elements=elements, key='added')`? Though not too complicated to understand as-is, if you prefer it this way :+1:

Reply:
By `?{updated|added}?` you mean `.getattr(delta_type)`?

BTW - key isn't `added` or `updated` .. it's `jobs`, `families` ...etc.

Yeah, I could, in two minds....

Reply:
>By ?{updated|added}? you mean .getattr(delta_type)?

I think so, was still thinking in JavaScript, where I could just access a property of an object as in an array element :smile: 

But no biggie on this one, code is working is well contained, so not a necessary change unless you want to :+1:

Reply:
Fixed